src/wasm/hostcall/mod.rs

```diff
@@ -95,50 +95,78 @@ mod wrapped_for_ffi {
     #[lucet_hostcall]
     #[no_mangle]
     pub extern "C" fn register(vmctx: &Vmctx, data: u32, length: u32) {
-        let internal_event = internal_events::Hostcall::begin(Role::Transform, "emit");
-        // TODO: Handle error.
-        let ret = super::register(vmctx, data, length).unwrap();
-        internal_event.complete();
-        ret
+        let internal_event = internal_events::WasmHostcallProgress::begin(Role::Transform, "emit");
+        match super::register(vmctx, data, length) {
+            Ok(_) => internal_event.complete(),
+            Err(e) => {
+                internal_event.error(format!("{}", e));
+            }
+        }
     }
 
     #[lucet_hostcall]
     #[no_mangle]
     pub extern "C" fn emit(vmctx: &Vmctx, data: u32, length: u32) -> u32 {
-        let internal_event = internal_events::Hostcall::begin(Role::Transform, "register");
-        // TODO: Handle error.
-        let ret = super::emit(vmctx, data, length).unwrap();
-        internal_event.complete();
-        ret
+        let internal_event =
+            internal_events::WasmHostcallProgress::begin(Role::Transform, "register");
```

Comment:
Looks like we have `"register"` being passed in `emit` and `"emit"` being passed in `register`.